Standardize special notes topic

hikchoi opened the pull request:
## What does this PR do?
<!-- Describe pull request here. -->

## What issues does this PR fix or reference?
<!-- Delete section or leave empty if no related GitHub Issue exists -->

- Fixes:
OR
- Relates to:

> **NOTE:** `Fixes` will close the ticket after merging. `Relates to:` will leave the ticket open after merging.

### Merge requirements satisfied

> For more information, visit the [Contributing Guide for Documentation](https://wiki.dendron.so/notes/b58801fc-43a9-4d42-a58b-eabc3e8538cb/)

- [ ] Check rendered output to ensure formatting is correct for renderings of wikilinks, note refs, tables, and images. `Dendron: Show Preview` can be used in your workspace to confirm the page renders as expected. Sometimes, `Dendron: Reload Index` needs to be ran if certain wikilinks aren't working as expected. (ignore this if contribution is made via the `Edit this page on GitHub` button from the published site)
- [ ] If this PR includes any refactoring (renaming notes, renaming headers, etc.), make sure that these changes were done via [Dendron Refactoring Commands](https://wiki.dendron.so/notes/srajljj10V2dl19nCSFiC/). This will ensure that any other impacted areas of the documentation, that link to or embed modified notes (via note references), are automatically updated.

Verify GitHub Actions tests are passing, which can be seen in the `Checks` tab of the PR:

- [ ] `URL validator` GitHub Action passes
- [ ] `Dendron site build` GitHub Action passes

> **NOTE:** If you are a new contributor, a maintainer will need to provide approval for GitHub Actions to run on your PRs.

kevinslin commented on vault/dendron.topic.special-notes.commands.create-daily-journal-note.md:
```diff
@@ -0,0 +1,19 @@
+---
+id: 9stxbnf9hxmxt76se5dtx2b
+title: "Dendron: Create Daily Journal Note"
+desc: "Create a daily journal note."
```
as a convention, can we not include the period in the frontmatter? if we want to add a period, we can add it afterwards

kevinslin commented on vault/dendron.topic.special-notes.commands.create-daily-journal-note.md:
```diff
@@ -0,0 +1,19 @@
+- mac: `cmd+shift+i`
+- windows: `ctrl+shift+i`
+
+## Details
```
i don't think details are necessary for this since its re-stating the title. on a second pass, we can show examples here

kevinslin commented on vault/dendron.topic.special-notes.md:
```diff
@@ -2,113 +2,71 @@
+They could also be creating using lookup by applying [[lookup modifiers|dendron.topic.lookup.modifiers]]
 
-## Scratch Note
+### Expand based on your use case
```
would modify to 

Builtin Best Practices

Special notes come with builtin schemas and templates. You can use them as is or customize for your own needs.

kevinslin commented on vault/dendron.topic.special-notes.meeting-note.md:
```diff
@@ -1,26 +1,38 @@
-## Components
+## Details
+
+### Use case
```
use cases have been a lvl2 header. can we keep it as such?

kevinslin commented on vault/dendron.topic.special-notes.meeting-note.md:
```diff
@@ -1,26 +1,38 @@
 {{fm.desc}}
 
-## Components
+## Details
```
remove this section if we don't use it

hikchoi replied:
I'm using it to bunch up everything below, but that may be excessive.
The past few doc standardizations I did use this categorization and if you want I can go back and remove them.

kevinslin reviewed (changes requested):
nice job! left some comments on wording. details and getting started around each of the note types look great!

hikchoi commented on vault/dendron.topic.special-notes.commands.create-journal-note.md:
```diff
@@ -0,0 +1,21 @@
+
+## Details
+
+**{{fm.title}}** is a command that lets you create a [[dendron.topic.special-notes.journal-note]].
+
```
I don't think we need to go that lean with the information it is providing. Every other command note has a details section starting with a sentence that tells you what it does in simple english, and without this sentence this note will stick out.

The top level outline of a command is:
- Summary: a one-liner description in simple english on what it does
- keybindings: keybindings if necessary
- details: everything else

This sentence being under details, there will be restatements happening frequently.
Right now this note really doesn't have a lot more detail so this sentence kind of sticks out to look redundant, but to preserve the completeness (in a sense that you can expect the same structure of information organized in every single note about a particular command), I would prefer this sentence to still be here.

hikchoi replied:
In short I'm arguing for redundancy for the sake of conformity in the standard we are applying. 

The content is sound without it, but our goal here is provide an expectation that any piece of information you get will be structured in a predictable way. I would say removing it will go against why we are doing this in the first place.

hikchoi replied:
Since this has been a common topic for discussion between us, let me go into my reasoning a bit more:

My thoughts on the argument for KISS / removing redundancy in documents.

The argument against keeping this sentence: 

- Imposing this structure of restating what the command does in the `details` section is unnecessary fluff, and an overhead that adds to document maintenance in the future.

The argument for keeping this sentence:

- That may be the case, and it would be beneficial if we are keeping this internally. 
- However, this is a public facing document.
  - With the assumption that readers want clear and predictable documentation, redundancy in documentation is not a concern.
  - In fact, I believe for the sake of keeping public facing documents presentable, the _writers should be the ones who bear the overhead if that is what the standard asks for_.
  - Let's say then, that as a standard that we make this sentence optional.
    - This also comes with its own overhead: _determining if this sentence should go here_.
    - By making it explicitly part of every single command note, we are eliminating this overhead of dealing with a tabula rasa of the `details` section, by suggesting that you start every command note details section with a longer-form description of what this command does.
- Standards by nature introduce redundancy.
  - Every piece of information we document can be reduced so that it contains all the necessary bits of information in the least amount of characters.
  - With standards, we are not doing that. 
    - We are not vacuum-packing information as air-tight as possible for storage, but rather packaging them in a nice box to be on a front window display.
- Bytes are cheap.
  - The redundancy has its purpose, and we are paying very little for that.

kevinslin replied:
so i think we should keep the details section. i'm saying we should cut the first sentence because it doesn't add new information. 

in terms of whats expensive, its not the bytes that are expensive but attention and time to read docs that don't add new information. (this was the big problem with javadocs - lots of text, but very little insight)

hikchoi replied:
I was under the impression that the template I created (which had the sentence) for past standardization are good to go.
My intention in the template was that each detail section starts with a sentence that reiterates what the command does;
preferably with a bit more meat to it so that it can provide a natural segue into what we are going to talk about in detail.

In the case of these commands there's not much to add for that piece of sentence, but I still value it being there because it serves as a common grounding point to every command note's detail section that you can later expand on.

We have came to a good agreement on the overall structure of each type of note, and here I'm trying to introduce some structure to each block of text so that it serves as some starting point for others when writing each section of a particular note consistently.

Well that was my intention, but I think we need more discussion on implementing that to the workflow, and that shouldn't block this PR.

For now (and in the future unless further discussions change this), I'll go with the following:

- If the command is so simple that the summary is enough to capture what it does, omit the detail section altogether.
- If there are things to add in the detail section, start the detail section with the opening sentence.

hikchoi replied:
I know we're constantly iterating on the standards as we go, and I apologize that the reviews turn into lengthy discussions sometimes.

Though without initiating conversation on these frictions during the review we're not going to end up with a solid standard at the end, and the workflow is going to incrementally become chaotic with each new ad-hoc decisions in the process.

I'll try to keep it short and come to a middle ground faster to make this process shorter, and async the conversations so they don't block

hikchoi replied:
> for the detail section, we can do a note reference to the detail section of the special note to explain more what this is doing

I'm not sure if that helps with

> (this was the big problem with javadocs - lots of text, but very little insight)

I would very much prefer to keep the details described in a command note to be directly related to the command and not go into the details of the output it creates which is already described elsewhere.

I'll omit the `## Details` section altogether if it makes no sense to have it.

hikchoi commented on vault/dendron.topic.special-notes.commands.create-meeting-note.md:
```diff
@@ -1,11 +1,18 @@
+## Keybindings
+none
+
+## Details
```
see: https://github.com/dendronhq/dendron-site/pull/616#discussion_r956893725

kevinslin replied:
same response as in #616. lets keep the detail section but add new information (transclude the special note special section)

hikchoi replied:
There was nothing relevant to transclude here in the context of the command itself, so added more details that pertains to the command.

hikchoi commented on vault/dendron.topic.special-notes.commands.create-scratch-note.md:
```diff
@@ -0,0 +1,21 @@
+
+## Details
+
+**{{fm.title}}** is a command that lets you create a [[dendron.topic.special-notes.scratch-note]].
```
see: https://github.com/dendronhq/dendron-site/pull/616#discussion_r956893725

hikchoi replied:
There was nothing relevant to transclude here in the context of the command itself, so added more details that pertains to the command.

hikchoi commented on vault/dendron.topic.special-notes.commands.create-task-note.md:
```diff
@@ -0,0 +1,18 @@
+
+## Details
+
+**{{fm.title}}** is a command that lets you create a task note.
```
see: https://github.com/dendronhq/dendron-site/pull/616#discussion_r956893725

hikchoi replied:
There was nothing relevant to transclude here in the context of the command itself, so added more relevant details.

hikchoi commented on vault/dendron.topic.special-notes.journal-note.md:
```diff
@@ -0,0 +1,33 @@
+
+### Getting Started
+
+![journal note](https://cdn.loom.com/sessions/thumbnails/3c3ddc1dc63547cea8bf186bec31f71b-with-play.gif)
```
I added a note in the spreadsheet we are tracking the process, but I'll add a comment here as well

hikchoi commented on vault/dendron.topic.special-notes.md:
```diff
@@ -2,113 +2,71 @@
 id: 5c213aa6-e4ba-49e8-85c5-1bdcb33ce202
 title: Special Notes
```
ack

hikchoi commented:
update:
- removed all trailing `.` from frontmatter and added them where it was substituted.
- removed the excessive sectioning with `details` and cleaned up the levels on each component / sub components
- added my thoughts on redundancy in relevant comments

hikchoi commented on vault/dendron.topic.special-notes.commands.create-daily-journal-note.md:
```diff
@@ -0,0 +1,19 @@
+## Summary
+
+{{fm.desc}}
+
```
Not sure what you mean here, are you saying the `## Summary` heading should link to the respective special note the command is creating?

kevinslin replied:
link to the summary section of those special notes. `creating a daily journal note` doesn't give us additional information whereas the summary of the special notes provide more detail about what those notes do

hikchoi commented:
update: 

- redundant sentence in `## Details` for each command note deleted
- More command related details added to `## Details` section
- Summary of special notes transcluded to the summary of the respective commands that create it
